exploration/quoted-literals.md

```diff
@@ -0,0 +1,403 @@
+- [r3 GOOD] Quotation marks are universally recognized as string delimiters.
+- [r4 FAIR] Quotation marks are not automatically paired by parsers nor IDEs,
+  but many text editors provide features to make working with and around quotes easier.
+
+### [a2] Dual quoting
+
+PR [#414](https://github.com/unicode-org/message-format-wg/pull/414) proposes to
+allow either single quotes `'` or double quotes `"` as literal delimiters,
+a variant of the "Use quotation marks" solution.
+
+- [r1 FAIR] Writing `"` and `'` in literals doesn't require escaping them via `\`,
+  as long as they do not match the literal's delimiter.
+  Literals containing both `'` and `"` will need to have at least one of those characters
+  escaped via `\`, which may itself need escaping in the container format.
+- [r2 GOOD] Embedding messages in certain container formats requires escaping the literal delimiters.
+  If the container format does not itself support dual quoting,
+  the embedded message's quotes may be adjusted to avoid their escaping.
+- [r3 GOOD] Quotation marks are universally recognized as string delimiters.
+- [r4 FAIR] Quotation marks cannot be paired by parsers nor IDEs,
+  but many text editors provide features to make working with and around quotes easier.
+
+### [a3] Use round or angle brackets
+
+- Round parentheses are very uncommon as string delimiters [r2 GOOD],
+  and thus may be surprising,
+  especially given the well-established meaning in prose [r4 POOR].
+  That said, there's prior art in using them for [delimiting strings in PostScript](https://en.wikipedia.org/wiki/PostScript#%22Hello_world%22).
+  Furthermore, they are relatively common in text, where they'd require escaping [r1 POOR].
+- Angle brackets require escaping in XML-based storage formats [r2 FAIR].
+- All brackets can be easily paired by parsers and IDEs [r5 GOOD].
+
+### [a4] Change escape introducer
+
+Changing the escape sequence introducer from backslash [c1] to another character
+could help partially mitigate the burden of first escaping literal delimiters
+and then escaping the escapes themselves [r1].
+However, it wouldn't address other requirements and use-cases.
+
+### [a5] Double delimiters to escape them
+
+This is the approach taken by ICU MessageFormat 1.0 for quotes.
+It allows literals to contain quotes [r1 GOOD]
+at the expense of doubling the amount of escaping required when embedding messages in code [r2 POOR].
+
+### [a6] Accept either `|` or quotes
+
+Allow any of the following as literal delimiters:
+
+- the vertical line character `|`
+- single quotes `'`
+- double quotes `"`
+
+This approach supports multiple different quoting styles to be used for literals.
+This flexibility allows for using a familiar and common style such as `'single'` or `"double"` quotes,
+while also allowing for `|pipes|` when the message's contents or embedding would otherwise require additional escaping.
```

Comment:
define "additional escaping". Do we mean due to the containing syntax? Or do we mean the literal's content? Or both?

Reply:
```suggestion
while also allowing for `|pipes|` when the message's contents or embedding would otherwise require additional escaping.
This means that literals could for example prefer `'single quotes'`,
but use `"double 'em"` if necessary,
or `|'pipe' characters|` if the whole message is wrapped in `"quotes"` due to the host format
or if the literal value contains both `'` and `"` quotes.
```